app/Http/Controllers/PhotoController.php

```diff
@@ -12,4 +14,107 @@ public function upload(Request $request)
     {
         return Inertia::render('Upload');
     }
+
+    public function store(Request $request, int $id)
+    {
+        // Data validation. A redirect is automatically done on invalid data.
+        $request->validate([
+            "images" => "required",
+            "images.*" => "image|mimes:jpg,png,jpeg|max:3000"
+        ]);
+
+        $files = $request->file('images');
+
+        $photosRows = array();
+
+        foreach($files as $file) {
+            $savedPath = $file->store("{$id}", 'album_data');
+
+            $exif = $this->exif_extract($file);
+
+            array_push($photosRows, [
+               "id_album" => $id,
+               "name" =>  $file->getClientOriginalName(),
+                "description" => "",
+                "filename" => $savedPath,
+                "latitude" => $exif["latitude"],
+                "longitude" => $exif["longitude"],
+                "date_p" => $exif["date_p"],
+                "created_at" => new DateTime(), // Theses two must be specified because Photo::insert doesn't use Eloquent
+                "updated_at" => new DateTime(), // Theses two must be specified because Photo::insert doesn't use Eloquent
+            ]);
+        }
+
+        // The insertion is done like that so we can let the database deal with the bulk insertion.
+        Photo::insert($photosRows);
+
+        return Redirect::route("album.gallery", ["id" => $id]);
+    }
+
+    /**
+     * Extract exif data (latitude, longitude, date) from a given picture
+     *
+     * @param file $file Picture to extract
```

Comment:
```suggestion
     * @param \Illuminate\Http\UploadedFile $file Picture to extract
```
The real class passed by parameter is an `UploadedFile`